bin/generate_npm_tarball.sh

```diff
@@ -1,28 +1,53 @@
 #!/bin/bash
 
 if [ $# -lt 2 ]; then
-  echo "usage: $0 PACKAGE[@VERSION] OUTPUT.TGZ [USE NODEJS 6]"
+  echo "usage: $0 PACKAGE[@VERSION] OUTPUT.TGZ [USE NODEJS 6] [SPECFILE_FOR_PURGE]"
   exit 1
 fi
 package=$1
 output=$(pwd)/$2
 usenodejs6=${3:-false}
+specfile="$4"
 wd=$(mktemp -d)
 trap "rm -rf '$wd'" EXIT INT TERM
 
 create_cache() {
   if [[ $usenodejs6 == true ]];then
     node_modules/npm/bin/npm-cli.js install --cache $wd/cache $package --no-shrinkwrap --no-optional --production --verbose
+    (cd $wd/cache && find -type f | sort) > $wd/to-compress
   else
+    if [[ -n $specfile ]] ; then
+      mkdir $wd/spec
+      spectool --get-files --directory $wd/spec "$specfile"
```

Comment:
Note: I also considered placing it in the output directory. That makes [this part](https://github.com/theforeman/foreman-packaging/blob/6a830812f7968d8477eced7ab933752dab0ec680/add_npm_package.sh#L64-L66) redundant. That command there also proved faster than spectool in the past. I'm not sure it's still needed or if spectool can also reuse connections now.